tests/unit/acquisition/function/test_greedy_batch.py

```diff
@@ -231,7 +231,10 @@ def test_fantasize_with_kriging_believer_does_not_change_negative_predictive_mea
 
     builder = Fantasizer(NegativePredictiveMean())
     acq0 = builder.prepare_acquisition_function(models, data)
-    acq1 = builder.prepare_acquisition_function(models, data, pending_points)
+
+    acq1 = builder.update_acquisition_function(acq0, models, data, pending_points[:1])
+    up_acq1 = builder.update_acquisition_function(acq1, models, data, pending_points)
+    assert up_acq1 == acq1  # in-place updates
```

Comment:
Not sure what's tested here. This test is a bit tricky because the `NegativePredictiveMean` does not change under Fantasize with Kriging believer. So this essentially checks that the equations are correct. With any other acquisition function acq1 and acq0 should differ. 

If you want to assess that `acq1 == up_acq1` in a general context, these lines should go in `test_fantasize_reduces_predictive_variance`. 

Sorry if I misunderstood something.